discovery-infra/tests/test_logs.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import tarfile
+import time
+
+from assisted_service_client.rest import ApiException
+from tests.base_test import BaseTest
+
+class TestDownloadLogs(BaseTest):
+    def setup_hosts(self, cluster_id, api_client, node_controller):
+        '''setup nodes from ISO image and wait until they are registered'''
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Boot nodes into ISO
+        node_controller.start_all_nodes()
+        # Wait until hosts are discovered and update host roles
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
+    
+    def prepare_for_installation(self, cluster_id, api_client, node_controller):
+        '''set roles and network params to prepare the cluster for install'''
+        self.set_host_roles(cluster_id=cluster_id, api_client=api_client)
+        self.set_network_params(cluster_id=cluster_id,
+                                 api_client=api_client,
+                                 controller=node_controller
+                                 )
+
+    def install_cluster_and_wait(self, cluster_id, api_client):
+        '''start the cluster and wait for host to go to installing-in-progress state'''
+        # Start cluster install
+        self.start_cluster_install(cluster_id=cluster_id, api_client=api_client)
+        # wait until all nodes are in Installed status, and the cluster moved to status installed
+        self.wait_for_cluster_to_install(cluster_id=cluster_id, api_client=api_client)
+
+    def test_collect_logs_on_success(self, api_client, node_controller, cluster):
+        # Define new cluster and prepare it for installation
+        cluster_id = cluster().id
+        self.setup_hosts(cluster_id, api_client, node_controller)
+        self.prepare_for_installation(cluster_id, api_client, node_controller)
+        
+        #download logs into a file. At this point logs are not uploaded
+        path = "/tmp/test_on-success_logs.tar"
+        self.verify_no_logs_uploaded(cluster_id, api_client, path)
+        
+        #install the cluster
+        ts_start_at = time.time()
+        self.install_cluster_and_wait(cluster_id, api_client)
+
+        #download logs into a file. At this point logs exist
+        expected_min_log_num = len(node_controller.list_nodes()) + 1
+        self.verify_logs_uploaded(cluster_id, api_client, path, ts_start_at, expected_min_log_num)
+
+    
+    def test_collect_logs_on_failure(self, api_client, node_controller, cluster):
+        '''cacnel insllation after at least one host is booted and check that logs are uploaded'''
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
```

Comment:
Why don't you use the previous functions you just defined? `setup_hosts` and `prepare_for_installation`